frontend/src/layouts/ProtectedLayout.vue

```diff
@@ -1,18 +1,55 @@
 <script setup lang="ts">
-import Header from "@/components/UI/Header.vue";
-import SideNav from "@/components/UI/SideNav.vue";
+import TermsAndConditions from "@/components/TermsAndConditions.vue";
+import Header from "@/layouts/Header.vue";
+import SideNav from "@/layouts/SideNav.vue";
+import Spinner from "@/components/UI/Spinner.vue";
+import { AppActlApiService } from "@/services/ApiServiceFactory";
+import { showTermsForAcceptance } from "@/store/TermsAndConditionsState";
+import { useQuery } from "@tanstack/vue-query";
+import { watch } from "vue";
+
+const termsAndConditionQuery = useQuery({
+    queryKey: ["user_terms_conditions", "user:validate"],
+    queryFn: () =>
+        AppActlApiService.userTermsAndConditionsApi
+            .validateUserRequiresAcceptTermsAndConditions()
+            .then((res) => res.data),
+});
+
+watch(termsAndConditionQuery.isSuccess, () => {
+    if (termsAndConditionQuery.data.value === true) {
+        showTermsForAcceptance();
+    }
+});
 </script>
 
 <template>
```

Comment:
This could benefit from a comment explaining the purpose of this component.

Reply:
https://github.com/bcgov/nr-forests-access-management/pull/1650/commits/796dd70b0c323ac5aeb029edeb4c2ad052e27d96